pkg/compiler/compiler_test.go

```diff
@@ -676,3 +676,202 @@ func Verify(a []byte, b []byte, s string, n int) bool {
 		})
 	}
 }
+
+func TestMapOKFlag(t *testing.T) {
+	tests := []struct {
+		name     string
+		src      string
+		expected any
+	}{
+		{
+			name: "int value, existing",
+			src: `
+				package foo
+				func Main() int {
+					var m = map[string]int{
+						"key": 1,
+					}
+					v, ok := m["key"]
+					if !ok {
+						panic("key is missing")
+					}
+					return v
+				}
+			`,
+			expected: big.NewInt(1),
+		},
+		{
+			name: "int value, missing",
+			src: `
+				package foo
+				func Main() int {
+					var m = make(map[string]int)
+					v, ok := m["unknown"]
+					if ok {
+						panic("key is existing")
+					}
+					return v
+				}
+			`,
+			expected: big.NewInt(0),
+		},
+		{
+			name: "slice value, existing",
+			src: `
+				package foo
+				func Main() []int {
+					var m = map[string][]int{
+						"key": []int{0, 1, 2},
+					}
+					var v, ok = m["key"]
+					if !ok {
+						panic("key is missing")
+					}
+					return v
+				}
+			`,
+			expected: []stackitem.Item{stackitem.Make(0), stackitem.Make(1), stackitem.Make(2)},
+		},
+		{
+			name: "slice value, missing",
+			src: `
+				package foo
+				func Main() []int {
+					var m = make(map[string][]int)
+					var v, ok = m["unknown"]
+					if ok {
+						panic("key is existing")
+					}
+					return v
+				}
+			`,
+			expected: nil,
+		},
+		{
+			name: "map value, missing",
+			src: `
+				package foo
+				func Main() map[string]int {
+					var m = make(map[string]map[string]int)
+					v, ok := m["unknown"]
+					if ok {
+						panic("key is existing")
+					}
+					return v
+				}
+			`,
+			expected: nil,
+		},
+		{
+			name: "struct value, missing",
+			src: `
+				package foo
+				type S struct {
+					A int
+				}
+				func Main() S {
+					var m = make(map[string]S)
+					var v, ok = m["key"]
+					if ok {
+						panic("key is existing")
+					}
+					return v
+				}
+			`,
+			expected: []stackitem.Item{stackitem.Make(0)},
+		},
+		{
+			name: "if with comma-ok idiom",
+			src: `
+				package foo
+				func Main() bool {
+					var m = map[string]int{
+						"key": 1,
+					}
+					if v, ok := m["key"]; !ok || v != 1{
+						return false
+					}
+					if _, ok := m["unknown"]; ok {
+						return false
+					}
+					return true
+				}
+			`,
+			expected: true,
+		},
+		{
+			name: "map returned from function, existing",
+			src: `
+				package foo
+				func Get() map[string]int {
+					return map[string]int{"key": 1}
+				}
+				func Main() int {
+					var v, ok = Get()["key"]
```

Comment:
One more test-case is:
```
				package foo
				var v int
				func getKey() string {
				    v = 5
				    return "key"
				}
				func Main() bool {
					var m = map[string]int{
						"key": 1,
					}
					_, _ = m[getKey()]
					return v
				}
```
It's needed to check that value of function is evaluated even if left-hand part of the expression is unused. It worked for existing type assertion code, and we need to check that the same situation is properly handled with maps.